### DISCUSSION TITLE
Month long battle to fix my printer. disclaimer, I have no clue wheat i'm doing

### DISCUSSION THREAD

**Discussion by akauder** [mriscoc/Ender3V2S1#1236] [Q&A]
To start, I have no clue what i'm doing, this is all new to me. I have an ender 3 v2 that recently had a board failure. I replaced the board with a skr mini e3 v3 and installed a bl touch. My first issue is firmware. I have attempted to compile firmware about 15 to 20 time using a variety of guides and setting and i have not gotten the printer to be fully functional. I have tried to compile using the custom firmware configurator with no luck due to issues flashing. My most recent attmpt was using the precompiled .bin file [Ender3V2-SKRME3V3-BLTUBL-MPC-20240125.bin] from (https://github.com/mriscoc/Ender3V2S1/releases/download/20240122/Ender3V2-SKRME3V3-BLTUBL-MPC-20240125.bin). this flashed and looked promising but came with lots of issues. 

All my issues seem to be related to auto leveling. I installed the firmware and I was able to use the probe wizard, adjust z offset and probe offset. I used the tramming wizard and i leveled the bed. I turned off the printer loaded my sd card with a print and turned it back on. I went ahead and used the auto home and the hot end drove into the bed without stopping. I killed the power and powered back on trying to attempt to recalabrate the probe. I start z probe wizard and it does the same thing and drives into the bed.

Ok, so now i start the z probe wizard and engage the probe early with my finger an half inch off the bed to make sure it stops. It stops and gives an error somthing like "printer killed reset power". I thought it strange that the probe stopped when i engaged higher but not when i let it touch the bed. To experiment I went ahead and raised the bed to it's maximum height and tried the z probe wizard again. Oddly enough the probe detected the bed and stopped and I was able to get the z offset configured but, now the bed is very uneven.

So i need to level the bed. I run the tramming wizard. and the hot end drives into the bed. I double check the z probe configuration. I use the z home wizard and the nozzle rams into the bed again on its initial probe but it stops then centers on the bed in the middle for the second probe and it stops perfectly. So I figure the bed is just very uneven, but even if that is the case why is the probe not working correctly in the tramming wizard and z probe wizard. it should be stopping before it hits the bed.  Why did the probe stop funtioning properly after a power cycle? Why does the bed height determine if the probe stops or drives the hotend ito the bed? Why does the nozzle drive into the bed when the bed is lower, but stop when the bed is raised up slightly? Due to this being a bin file I cannot edit the configuration so i once again tried to build and flash a new firmware file, but now i cannot get any firmware to flash. I hope someone has some advice for me. 

I dont need anything fancy i just want a functioning printer with auto bed leveling. If someone has a usable firmware for their ender 3 v2 with skr mini e3 v3 and bl touch i would very much appreciate it. Yes i know that there are firmware files on here and i have tried them. I have read through the forums i have been at this for weeks on and off. Please give me a break if i have missed something in the forums. thanks for the help in advance.

Can someone please adivise on what im doing wrong?

**Comment by akauder** [comment]
Update. i got it to stop hitting the bed most of the time. I re-leveld the bed manually and then recalibrated the z offset. first issue is, it still hits the bed once in a while for some odd reason. I also have had some strange bugs where the hot end fan goes berserk and tries to runaway. 
I was able to get a print running today. I am having an issue getting the bed to level before each print. I added the G29 command in cura but it does not check the bed before the print. I am not sure what i am doing wrong there. I also have a strange issue where the print is set to 200c in cura but when i go to start print it heats the bed but keeps the hot end at 0. I have to exit the print and either retry or i run the prepare pla preset (set to 195) and then start the print and it works. Im not certain what specifically fixes it.

**Reply by mriscoc** [reply]
Maybe you can find useful information here: https://github.com/mriscoc/Ender3V2S1/discussions/131
Also you could ask to other skr mini e3 v3 users, because your board issues are not from normal behavior.

**Comment by akauder** [comment]
I appreciate that. I think I may end up replacing my board again as I think
it may have a short of some kind due to the erratic behavior.
